iOS/DuckDuckGo/AppLifecycle/AppStates/Foreground.swift

```diff
@@ -121,7 +122,7 @@ struct Foreground: ForegroundHandling {
         services.dbpService.resume()
         services.inactivityNotificationSchedulerService.resume()
         services.widePixelService.resume()
-
+        services.launchSourceManager.handleAppAction(launchAction)
```

Comment:
My only concern is the placement of this call.

It's breaking the unwritten convention of a nice resume call list in this function. I'd probably move the `LaunchSourceManager` to be a dependency of `UIInteractionManager`, which (citing the code doc) _orchestrates the following operations: (...) 3. Handles immediate launch actions (if any) (...)_. 
With this the source storing can be done in a way that makes sure the source is available before the UI is being shown. 

cc @jaceklyp, any thoughts? Maybe I'm overthinking this?